lib/internal/modules/run_main.js

```diff
@@ -80,9 +81,23 @@ function shouldUseESMLoader(mainPath) {
 function runMainESM(mainPath) {
   const { loadESM } = require('internal/process/esm_loader');
   const { pathToFileURL } = require('internal/url');
+  const _importMapPath = getOptionValue('--experimental-import-map');
   const main = pathToFileURL(mainPath).href;
 
   handleMainPromise(loadESM((esmLoader) => {
+    // Load import map and throw validation errors
+    if (_importMapPath) {
+      const { ImportMap } = require('internal/modules/esm/import_map');
+      const { getCWDURL } = require('internal/util');
+
+      const importMapPath = esmLoader.resolve(_importMapPath, getCWDURL(), { __proto__: null, type: 'json' });
+      return esmLoader.import(importMapPath.url, getCWDURL(), { __proto__: null, type: 'json' })
+        .then((importedMapFile) => {
+          esmLoader.importMap = new ImportMap(importedMapFile.default, pathToFileURL(path.dirname(new URL(importMapPath.url).pathname)));
```

Comment:
cc @aduh95 @JakobJingleheimer @GeoffreyBooth 

This is what I ended up with after trying a few differnt ways to get this to work. Maybe there are some other api's I dont know about which would help here? I could not see an easy way to use the loader to do this while also providing the resolved path. And then because we need the directory to resolve against as the base url and what we get back is a file path I did not see a good way (other than sting manipulation) to pass a url to `path.dirname`. Again, I am pretty sure I am missing something here because does not seem like it should be this complicated.